### PR TITLE
Frontend setup

### PR DESCRIPTION
**Pull request by aka-jain**
Initiallizing frontend directory for EvalAI UI code base. Integrating with Gulp, SASS and Angular.

### CODE REVIEW

**Comment by dexter1691** [comment]
Are we planning to serve the files using Django or Node.js. I thought @deshraj said that serving the files using Django will be easier. Serving with Node.js will mean that we have to run things on two different ports and build a reverse proxy. Not sure we want to do that.

**Comment by deshraj** [comment]
@dexter1691 I think we agreed on having a django server for REST APIs and a nodejs server running on different port which will consume APIs, no? 

This will keep frontend decoupled from backend. Let me know if you have concerns.

**Comment by dexter1691** [comment]
I see. How will it run on AWS. Will it run on same server? Or on different server?
If same server: how do you deal with reverse proxy
If different server: we will have to scale / manage two servers.

**Comment by deshraj** [comment]
Ammm...Good point. Earlier I was planning to go with the first method i.e setting both django and nodejs on the same EBS env. 

Setting the reverse proxy would be a bit complex on EBS and running two projects in the same EBS enviroment doesn't seem like a good idea. We can serve everything using django only(Involves less pain).

File: frontend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# files/directories to be ignored
+
+.sass-cache/
+
+node_modules/
+bower_components/
```
Comment (deshraj) [inline frontend/.gitignore]: @aka-jain can you add a newline here ?

File: frontend/index.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+<!-- evalAI main scripts -->
+<script type="text/javascript" src="dist/assets/js/app.js"></script>
+<script type="text/javascript" src="dist/assets/js/controllers.js"></script>
+<script type="text/javascript" src="dist/assets/js/directives.js"></script>
+<!-- <script type="text/javascript" src="dist/assets/js/services.js"></script> -->
+
+
+</body>
+</html>
```
Comment (deshraj) [inline frontend/index.html]: Please add a newline here also.

Comment (aka-jain) [inline frontend/index.html]: Sure @deshraj. Will do it.

File: frontend/.gitignore (hunk 1 above)
Comment (aka-jain) [inline frontend/.gitignore]: Sure.

**Comment by dexter1691** [comment]
@aka-jain: Plz make the necessary changes and merge the master branch to approve the PR. Thanks!

**Comment by aka-jain** [comment]
@dexter1691 : sure, will do it as soon as I complete the setup

**Comment by aka-jain** [comment]
Closing this PR and creating a new one as there is a huge change in UI code base.

**Pull request closed without merging**